src/lib/utils/ct_utils.h

```diff
@@ -134,6 +135,10 @@ class Choice final {
          return Choice(ct_is_zero<uint32_t>(static_cast<uint32_t>(v_is_0)));
       }
 
+      constexpr static Choice yes() { return Choice::from_int(1U); }
+
+      constexpr static Choice no() { return Choice::from_int(0U); }
```

Comment:
These could just directly initialize the masked values (either all zero or all ones resp), no need for them to be guarded by the value barrier.

Reply:
Given that the constexpr-ness of the factory those delegate to, I was hoping that the compiler would do that for me and not actually compile a value barrier. But yeah, no harm in doing that explicitly.

Reply:
That would only apply if you created the variable as `constexpr` though, right?

```
constexpr auto yes = Choice::yes(); // compile time evaluated, no value barrier
const auto yes = Choice::yes(); // evaluted at runtime, presumably with value barrier
```